src/bin/pg_autoctl/cli_service.c

```diff
@@ -210,27 +210,10 @@ cli_monitor_run(int argc, char **argv)
 	pg_setup_get_local_connection_string(&(mconfig.pgSetup), connInfo);
 	monitor_init(&monitor, connInfo);
 
-	if (!pg_is_running(mconfig.pgSetup.pg_ctl, mconfig.pgSetup.pgdata))
-	{
-		log_info("Postgres is not running, starting postgres");
 
-		if (!pg_ctl_start(mconfig.pgSetup.pg_ctl,
-						  mconfig.pgSetup.pgdata,
-						  mconfig.pgSetup.pgport,
-						  mconfig.pgSetup.listen_addresses))
-		{
-			log_error("Failed to start PostgreSQL, see above for details");
-			exit(EXIT_CODE_PGCTL);
-		}
-	}
-
-	log_info("PostgreSQL is running in \"%s\" on port %d",
-			 mconfig.pgSetup.pgdata, mconfig.pgSetup.pgport);
-
-	/* Check version compatibility */
-	if (!monitor_ensure_extension_version(&monitor, &version))
+	if (!ensure_monitor_pg_running(&monitor, &mconfig))
```

Comment:
I think this call should go in `monitor_service_run` itself. It would be easier to follow the logic, because that's where we connect to the local Postgres too.